pyam/ixmp4.py

```diff
@@ -70,6 +70,48 @@ def read_ixmp4(
     return IamDataFrame(data, meta=meta, index=index)
 
 
+def read_ixmp4_run(
+    run: ixmp4.Run,
+    region: str | list[str] | None = None,
+    variable: str | list[str] | None = None,
+    unit: str | list[str] | None = None,
+    year: int | list[int] | None = None,
+):
+    """Return timeseries data and meta indicatos from an ixmp4 run as IamDataFrame
+
+    Parameters
+    ----------
+    run : :class:`ixmp4.Run`
+        An ixmp4 run instance from which to read timeseries data and meta indicators
+    region, variable, unit : str or list of str, optional
+        Filter timeseries data by these dimensions.
+    year : int or list of int, optional
+        Filter timeseries data by time domain.
+    """
+    from pyam import IamDataFrame
```

Comment:
Is there a special reason to not put the import at the top of the file?

Reply:
Ah ok, maybe there's a good way to rearrange some code to get rid of the circular dependency but that's a job for another day.